### PR TITLE
ENH: add has_m

### PR DESCRIPTION
**Pull request by mwtoews**
Add a basic `has_m` unary predicate function, available with GEOS >= 3.12.0.

#### Other changes to note
* Update some related docs for `get_coordinate_dimension`
* Add `has_z` to `UNARY_PREDICATES` for shapely/tests/test_predicates.py

### CODE REVIEW

File: shapely/tests/test_io.py [hunk 1]
```diff
@@ -1095,7 +1095,8 @@ def test_pickle_m(geom):
     actual = pickle.loads(pickled)
     assert_geometries_equal(actual, geom, tolerance=0)
     assert not actual.has_z
-    # TODO: assert actual.has_m
+    if not actual.is_empty:  # GEOSHasM with EMPTY geometries is inconsistent
```
Comment (jorisvandenbossche) [inline shapely/tests/test_io.py:1098]: There is an upstream issue about this? 
Or is for the case of multi empty geometries (https://github.com/libgeos/geos/issues/888)?

File: shapely/tests/test_predicates.py [hunk 2]
```diff
@@ -218,17 +227,47 @@ def test_dwithin():
...
+    assert not shapely.has_z(geometry)
+    assert shapely.has_m(geometry)
+
+
+@pytest.mark.skipif(
+    shapely.geos_version < (3, 12, 0),
+    reason="M coordinates not supported with GEOS < 3.12",
+)
+@pytest.mark.parametrize("geometry", all_types_zm)
+def test_has_m_all_types_zm(geometry):
```
Comment (jorisvandenbossche) [inline shapely/tests/test_predicates.py:266]:
```suggestion
def test_has_z_has_m_all_types_zm(geometry):
```

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

File: shapely/tests/test_io.py (hunk 1 above)
Comment (mwtoews) [reply shapely/tests/test_io.py:1098]: That's the main issue, and there are other GEOS-version-specific issues related to GEOSHasZ too like https://github.com/libgeos/geos/pull/887. These will hopefully get less complicated when support for older GEOS versions drops off. There are a few other "EMPTY geometries is inconsistent" comments that I've added to `main` with the same theme.

**Pull request merged**